app/main.py

```diff
@@ -1,8 +1,30 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        self.people[name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    def __repr__(self) -> str:
+        return f"<Person name={self.name}>"
+
+
+def create_person_list(people_data: list) -> list:
+    person_list = []
+    for person_dict in people_data:
+        name = person_dict["name"]
+        age = person_dict["age"]
+        person = Person(name=name, age=age)
+        if "wife" in person_dict and person_dict["wife"] is not None:
+            wife_name = person_dict["wife"]
+            if wife_name in Person.people:
+                person.wife = Person.people[wife_name]
+                Person.people[wife_name].husband = person
```

Comment:
Due to task if "wife" inside person you must set only wife, noting said about husband